src/HotChocolate/Language/src/Language.SyntaxTree/NamedSyntaxNode.cs

```diff
@@ -116,11 +116,9 @@ public override bool Equals(object? obj)
     /// </returns>
     public override int GetHashCode()
     {
-        unchecked
-        {
-            var hashCode = Name.GetHashCode();
-            hashCode = (hashCode * 397) ^ EqualityHelper.GetHashCode(Directives);
-            return hashCode;
-        }
+        var hashCode = new HashCode();
```

Comment:
In the hash code we always need o make sure the kind is included. We do not need the kind on the equals since the equality method gets a NamedSyntax node but on the hash this improves the hashcode and we potentially have less collisions.

Reply:
Actually you can leave these ... I already fixed these in my local branch. Just as an info.